system/jlib/jptree.cpp

```diff
@@ -3132,6 +3132,7 @@ void PTree::addLocal(size32_t l, const void *data, bool _binary, int pos)
         IPropertyTree *element1 = detach();
         array = new CPTArray();
         addingNewElement(*element1, ANE_APPEND);
+        static_cast<PTree *>(element1)->setOwner(array);
```

Comment:
Without this YAML generation and merging don't work correctly.  I suspect you may want a dynamic cast instead of a static cast....